akka-actor-typed/src/main/scala/akka/actor/typed/receptionist/Receptionist.scala

```diff
@@ -269,15 +281,72 @@ object Receptionist extends ExtensionId[Receptionist] {
+     * for a service key.
+     *
+     * For a clustered `ActorSystem` it only contain services on nodes that
+     * are not seen as unreachable (note that they could have still have become
+     * unreachable between this message being sent and the receiving actor processing it).
+     *
+     * For a list including both reachable and unreachable instances see [[#allServiceInstances]]
      *
      * Also, see [[ServiceKey.Listing]] for more convenient pattern matching
      */
     def serviceInstances[T](key: ServiceKey[T]): Set[ActorRef[T]]
 
-    /** Java API */
+    /**
+     * Java API: Return the reachable service instances.
+     *
+     * In a non-clustered `ActorSystem` this will always be all registered instances
+     * for a service key.
+     *
+     * For a clustered `ActorSystem` it only contain services on nodes that has
+     * are not seen as unreachable (note that they could have still have become
+     * unreachable between this message being sent and the receiving actor processing it).
+     *
+     * For a list including both reachable and unreachable instances see [[#getAllServiceInstances]]
+     */
     def getServiceInstances[T](key: ServiceKey[T]): java.util.Set[ActorRef[T]]
 
+    /**
+     * Scala API: Return both the reachable and the unreachable service instances.
+     *
+     * In a non-clustered `ActorSystem` this will always be the same as [[#serviceInstances]].
+     *
+     * For a clustered `ActorSystem` this include both services on nodes that are reachable
+     * and nodes that are unreachable.
+     */
+    def allServiceInstances[T](key: ServiceKey[T]): Set[ActorRef[T]]
+
+    /**
+     * Java API: Return both the reachable and the unreachable service instances.
+     *
+     * In a non-clustered `ActorSystem` this will always be the same as [[#getServiceInstances]].
+     *
+     * For a clustered `ActorSystem` this include both services on nodes that are reachable
+     * and nodes that are unreachable.
+     */
+    def getAllServiceInstances[T](key: ServiceKey[T]): java.util.Set[ActorRef[T]]
+
+    /**
+     * Scala API: `true` only if new services were added or removed and `false` if this listing is
+     * only about reachability changes. Useful for subscribers only concerned with [[#allServiceInstances]].
+     *
+     * In a non-clustered `ActorSystem` this will be `true` for all listings.
+     * For `Find` queries and the initial listing for a `Subscribe` this will always be `true`.
+     */
+    def allServiceInstancesChanged: Boolean
+
+    /**
+     * JavaAPI: `true` only if new services were added or removed and `false` if this listing is
+     * only about reachability changes. Useful for subscribers only concerned with [[#getAllServiceInstances]].
+     *
+     * In a non-clustered `ActorSystem` this will be `true` for all listings.
+     * For `Find` queries and the initial listing for a `Subscribe` this will always be `true`.
+     */
+    def getAllServiceInstancesChanged: Boolean
```

Comment:
I think normal Java convention is to use `isX` for Boolean

Reply:
I actually think we can have one `isAllServiceInstancesChanged` for both Scala and Java.

For some reason I think `allServiceInstancesChanged` can be misinterpreted as "all instances has changed"

Reply:
Perhaps inverting and calling it "onlyReachableChanged" or instead?

Reply:
BTW the separate method was mostly to link to the right `allServiceInstances` method from the scaladoc

Reply:
I think it's good that we don't tie this to closely to reachability/unreachability. That way we could have other aspects for choosing preferred vs all in the future. E.g. other health or load metrics.